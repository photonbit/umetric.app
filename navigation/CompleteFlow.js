import React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import {
  createDrawerNavigator,
  DrawerContentScrollView,
  DrawerItemList,
} from '@react-navigation/drawer'
import i18n from 'i18n-js'

import { BasicInputFlow } from './InputFlow'
import { EditFlow } from './EditFlow'
import { GoalsFlow } from './GoalsFlow'
// import { QuestionnaireFlow } from "./QuestionnaireFlow"
import { navigationRef } from './RootNavigation'

const Drawer = createDrawerNavigator()

export default function CompleteFlow() {
  return (
    <NavigationContainer ref={navigationRef}>
      <Drawer.Navigator
        drawerContent={(props) => {
          return (
            <DrawerContentScrollView {...props}>
              <DrawerItemList {...props} />
            </DrawerContentScrollView>
          )
        }}
        screenOptions={{
            headerShown: false,
            swipeEdgeWidth: 300,
        }}
      >
        <Drawer.Screen
          name="Input"
          component={BasicInputFlow}
          options={{ title: i18n.t('register') }}
        />
        <Drawer.Screen name="Edit" component={EditFlow} options={{ title: i18n.t('edit') }} />
        <Drawer.Screen name="Metas" component={GoalsFlow} options={{ title: i18n.t('goals') }} />
        {/* Disable QuestionnaireFlow for the first version
        <Drawer.Screen name="Cuestionarios" component={QuestionnaireFlow} options={{ title: i18n.t('questionnaires') }} />
        */}
      </Drawer.Navigator>
    </NavigationContainer>
  )
}
