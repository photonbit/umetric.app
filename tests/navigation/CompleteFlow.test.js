import React from 'react';
import renderer from 'react-test-renderer';

import CompleteFlow from '../../navigation/CompleteFlow';

describe('<CompleteFlow />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<CompleteFlow />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});

it('renders correctly', () => {
  const tree = renderer.create(<CompleteFlow />).toJSON();
  expect(tree).toMatchSnapshot();
});