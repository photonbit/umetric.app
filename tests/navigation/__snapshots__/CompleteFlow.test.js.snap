// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                uMetric
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "white",
                          "flex": 1,
                          "justifyContent": "space-evenly",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "flexWrap": "wrap",
                          }
                        }
                      >
                        <Image
                          resizeMode="contain"
                          source={1}
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      >
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          placeholder="Usuario..."
                          placeholderTextColor="#003f5c"
                          rejectResponderTermination={true}
                          style={
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "borderColor": "#CCCCCC",
                              "borderRadius": 5,
                              "borderWidth": 1,
                              "color": "#111111",
                              "fontSize": 18,
                              "height": 50,
                              "marginBottom": 25,
                              "marginTop": 15,
                              "paddingBottom": 5,
                              "paddingLeft": 10,
                              "paddingRight": 10,
                              "paddingTop": 5,
                              "width": "100%",
                            }
                          }
                          textContentType="username"
                          underlineColorAndroid="transparent"
                          value=""
                        />
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          placeholder="Contraseña..."
                          placeholderTextColor="#003f5c"
                          rejectResponderTermination={true}
                          secureTextEntry={true}
                          style={
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "borderColor": "#CCCCCC",
                              "borderRadius": 5,
                              "borderWidth": 1,
                              "color": "#111111",
                              "fontSize": 18,
                              "height": 50,
                              "marginBottom": 25,
                              "marginTop": 15,
                              "paddingBottom": 5,
                              "paddingLeft": 10,
                              "paddingRight": 10,
                              "paddingTop": 5,
                              "width": "100%",
                            }
                          }
                          textContentType="password"
                          underlineColorAndroid="transparent"
                          value=""
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#48BBEC",
                              "borderRadius": 15,
                              "justifyContent": "space-evenly",
                              "opacity": 1,
                              "padding": 15,
                              "width": 200,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 24,
                              }
                            }
                          >
                            Acceder
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
